cmd/storage-rpc-client.go

```diff
@@ -115,20 +128,33 @@ func newStorageRPC(ep *url.URL) (StorageAPI, error) {
 		}
 	}
 
-	storageAPI := &networkStorage{
-		rpcClient: newAuthRPCClient(authConfig{
-			accessKey:        accessKey,
-			secretKey:        secretKey,
-			serverAddr:       rpcAddr,
-			serviceEndpoint:  rpcPath,
-			secureConn:       isSSL(),
-			serviceName:      "Storage",
-			disableReconnect: true,
-		}),
+	netStorage := &networkStorage{}
+	netStorage.rpcClient = newAuthRPCClient(authConfig{
+		accessKey:       accessKey,
+		secretKey:       secretKey,
+		serverAddr:      rpcAddr,
+		serviceEndpoint: rpcPath,
+		secureConn:      isSSL(),
+		serviceName:     "Storage",
+	})
+	netStorage.retryCfg = rCfg
+	netStorage.verifyServerFunc = func() error {
+		// FIXME: This is basic server validation check by reading format.json.
+		//        However there could be many possible misconfiguration currently
+		//        we don't handle.
+		// Read format config file to validate the server.
+		buf, err := netStorage.readAll(minioMetaBucket, formatConfigFile, nil)
```

Comment:
The previous code it wasn't possible but it is important since you would want a clean state of StorageAPI. Also additional benefit is to avoid mutation inside a callback.